src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -571,20 +575,29 @@ private void ExecuteSyntaxTreeActions(AnalysisScope analysisScope, AnalysisState
                     continue;
                 }
 
-                foreach (var analyzer in analysisScope.Analyzers)
+                var processedAnalyzers = analysisStateOpt != null ? PooledHashSet<DiagnosticAnalyzer>.GetInstance() : null;
```

Comment:
is this micro op worth it?why not always just hav ethis set?

Reply:
This is hot path as dictionary accesses/allocations here showed up in the trace. This set is never required in command line builds as `analysisStateOpt` is only ever non-null for partial analysis, i.e. IDE open file analysis. I believe it is worth avoiding allocating it here.